File: Frontend/LeaveManagement/src/app/features/Attendance/Dtos/EmployeeAttendanceResponseDto.ts
export interface EmployeeAttendanceResponseDto {
    clockOut(clockOut: any): unknown;
    clockIn(clockIn: any): unknown;
    date: string; // DateOnly maps to string (e.g., 'YYYY-MM-DD')
    employeeId: number;
    firstName?: string; // string? in C# maps to optional string in TypeScript
    lastName?: string;  // string? in C# maps to optional string in TypeScript
    clockInTime: Date;
    clockOutTime: Date;
    workHours: number;
    percentage: number;
  }

File: Backend/LeaveManagement/LeaveManagement/Application/DTOs/EmployeeAttendanceResponseDTO.cs
namespace LeaveManagement.Application.DTOs;
public class EmployeeAttendanceResponseDTO
{
    public DateOnly Date { get; set; }
    public int EmployeeId { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public DateTime ClockInTime { get; set; }
    public DateTime ClockOutTime { get; set; }
    public double WorkHours { get; set; } 
    public double Percentage { get; set; }

}
